ompi/mca/coll/libnbc/nbc_ialltoall.c

```diff
@@ -544,3 +586,27 @@ static inline int a2a_sched_inplace(int rank, int p, NBC_Schedule* schedule, voi
 
   return OMPI_SUCCESS;
 }
+
+int ompi_coll_libnbc_alltoall_init (const void* sendbuf, int sendcount, MPI_Datatype sendtype, void* recvbuf, int recvcount,
+                                    MPI_Datatype recvtype, struct ompi_communicator_t *comm, MPI_Info info, ompi_request_t ** request,
+                                    struct mca_coll_base_module_2_2_0_t *module) {
+    int res = nbc_ialltoall(sendbuf, sendcount, sendtype, recvbuf, recvcount, recvtype,
```

Comment:
If the intention for this function is to be the common initialisation routine for persistent and nonblocking all-to-all, then it would be better to name it ```nbc_alltoall_init``` (rather than ```nbc_ialltoall```). Future programmers will be less easily confused by that name, and less likely to (re-)introduce data-movement code into the initialisation-only step. Also, the nonblocking code path will then read correctly (initialise via ```nbc_alltoall_init``` then start via ```NBC_Start```).

Reply:
Have a point. I'll fix it.